assets/cubyz/shaders/skybox/fragment.fs

```diff
@@ -0,0 +1,179 @@
+#version 330 
+
+layout (location=0) out vec4 fragColor;
+
+in vec3 pos;
+
+#define PI 3.141592
+#define iSteps 16
+#define jSteps 8
+#define ATMOS_SIZE 6371e3
+
+uniform mat4 lightDir;
+uniform mat4 invLightDir;
+uniform float altitude;
+
+vec2 rsi(vec3 r0, vec3 rd, float sr) {
+	float a = dot(rd, rd);
+	float b = 2.0 * dot(rd, r0);
+	float c = dot(r0, r0) - (sr * sr);
+	float d = (b*b) - 4.0*a*c;
+	if (d < 0.0) return vec2(1e5,-1e5);
+	return vec2(
+		(-b - sqrt(d))/(2.0*a),
+		(-b + sqrt(d))/(2.0*a)
+	);
+}
+
+vec3 atmosphere(vec3 r, vec3 r0, vec3 pSun, float iSun, float rPlanet, float rAtmos, vec3 kRlh, float kMie, float shRlh, float shMie, float g) {
+	pSun = normalize(pSun);
+	r = normalize(r);
+
+	vec2 p = rsi(r0, r, rAtmos);
+	if (p.x > p.y) return vec3(0,0,0);
+	p.y = min(p.y, rsi(r0, r, rPlanet).x);
+	float iStepSize = (p.y - p.x) / float(iSteps);
+
+	float iTime = 0.0;
+
+	vec3 totalRlh = vec3(0,0,0);
+	vec3 totalMie = vec3(0,0,0);
+
+	float iOdRlh = 0.0;
+	float iOdMie = 0.0;
+
+	float mu = dot(r, pSun);
+	float mumu = mu * mu;
+	float gg = g * g;
+	float pRlh = 3.0 / (16.0 * PI) * (1.0 + mumu);
+	float pMie = 3.0 / (8.0 * PI) * ((1.0 - gg) * (mumu + 1.0)) / (pow(1.0 + gg - 2.0 * mu * g, 1.5) * (2.0 + gg));
+
+	for (int i = 0; i < iSteps; i++) {
+
+		vec3 iPos = r0 + r * (iTime + iStepSize * 0.5);
+
+		float iHeight = length(iPos) - rPlanet;
+
+		float odStepRlh = exp(-iHeight / shRlh) * iStepSize;
+		float odStepMie = exp(-iHeight / shMie) * iStepSize;
+
+		iOdRlh += odStepRlh;
+		iOdMie += odStepMie;
+
+		float jStepSize = rsi(iPos, pSun, rAtmos).y / float(jSteps);
+
+		float jTime = 0.0;
+
+		float jOdRlh = 0.0;
+		float jOdMie = 0.0;
+
+		for (int j = 0; j < jSteps; j++) {
+
+			vec3 jPos = iPos + pSun * (jTime + jStepSize * 0.5);
+
+			float jHeight = length(jPos) - rPlanet;
+
+			jOdRlh += exp(-jHeight / shRlh) * jStepSize;
+			jOdMie += exp(-jHeight / shMie) * jStepSize;
+
+			jTime += jStepSize;
+		}
+
+		vec3 attn = exp(-(kMie * (iOdMie + jOdMie) + kRlh * (iOdRlh + jOdRlh)));
+
+		totalRlh += odStepRlh * attn;
+		totalMie += odStepMie * attn;
+
+		iTime += iStepSize;
+
+	}
+
+	return iSun * (pRlh * kRlh * totalRlh + pMie * kMie * totalMie);
+}
+
+vec3 hash( vec3 x )
+{
+	x = vec3(
+		dot(x,vec3(127.1,311.7, 74.7)),
+		dot(x,vec3(269.5,183.3,246.1)),
+		dot(x,vec3(113.5,271.9,124.6))
+	);
+
+	return fract(sin(x)*43758.5453123);
+}
+
+vec3 voronoi_sphere( in vec3 x )
```

Comment:
This function produces rather boring stars, since they are all uniformly distributed on a sphere.
(technically they are not even uniform since I can see artifacts everywhere)

Also I think that rasterization is the better solution here:
Instead of looking for every pixel, which star is closest, you can just draw each star individually.
This allows for arbitrary precomputed star patterns (how about a kind of volumetric disk, similar to the milky way?)